Lib/test/test_bytes.py

```diff
@@ -1599,6 +1599,10 @@ def test_extend(self):
         a = bytearray(b'')
         a.extend([Indexable(ord('a'))])
         self.assertEqual(a, b'a')
+        a = bytearray(b'abc')
+        self.assertRaisesRegex(TypeError,
+                               "Expected iterable of integers; got: 'str'",
+                               a.extend, 'def')
```

Comment:
I decided to add assertion that do not always override TypeError.
Also, messages begin with lowercase (I checked several others, including [].extend)
since printed after uppercase exception class.
```suggestion
        self.assertRaisesRegex(TypeError,  # Override for string.
                               "expected iterable of integers; got: 'str'",
                               a.extend, 'def')
        self.assertRaisesRegex(TypeError,  # But not for others.
                               "can't extend bytearray with float",
                               a.extend, 1.0)
```